http/src/main/java/io/micronaut/http/filter/FilterProperties.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright 2017-2019 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.http.filter;
+
+import io.micronaut.core.annotation.AnnotationValue;
+import io.micronaut.core.annotation.Internal;
+import io.micronaut.core.util.ArrayUtils;
+import io.micronaut.http.HttpMethod;
+import io.micronaut.http.annotation.Filter;
+
+import java.lang.annotation.Annotation;
+
+/**
+ * Encapsulates all the possible configurations that might be defined in {@link Filter} annotation.
+ */
+@Internal
+public final class FilterProperties {
```

Comment:
If this is public API then remove `@Internal` and make sure any public methods are marked with `@Nonnull` or `@Nullable`